core/src/test/java/io/grpc/internal/ClientCallImplTest.java

```diff
@@ -147,6 +155,130 @@ public void tearDown() {
   }
 
   @Test
+  public void exceptionInOnMessageTakesPrecedenceOverServer() {
+    DelayedExecutor executor = new DelayedExecutor();
+    ClientCallImpl<Void, Void> call = new ClientCallImpl<Void, Void>(
+        method,
+        executor,
+        CallOptions.DEFAULT,
+        provider,
+        deadlineCancellationExecutor);
+    call.start(callListener, new Metadata());
+    verify(stream).start(listenerArgumentCaptor.capture());
+    final ClientStreamListener streamListener = listenerArgumentCaptor.getValue();
+    streamListener.headersRead(new Metadata());
+
+    RuntimeException failure = new RuntimeException("bad");
+    doThrow(failure).when(callListener).onMessage(any(Void.class));
+    final AtomicBoolean cancelCalled = new AtomicBoolean();
+    doAnswer(new Answer<Void>() {
+      @Override
+      public Void answer(InvocationOnMock invocation) throws Throwable {
+        // The real stream will ignore extra cancel calls, so no need to reinvoke the listener.
+        cancelCalled.set(true);
+        return null;
+      }
+    }).when(stream).cancel(any(Status.class));
```

Comment:
Did you want to make this a normal verify? Mainly because it is copied for each of these tests.